vault/resource_pki_backend_role.go

```diff
@@ -0,0 +1,380 @@
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for server use.",
+			},
+			"client_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for client use.",
+			},
+			"code_signing_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for code signing use.",
+			},
+			"email_protection_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for email protection use.",
+			},
+			"key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "rsa",
+				Description: "Specifies the type of key to generate for generated private keys. Currently, rsa and ec are supported.",
+			},
+			"key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     2048,
+				Description: "Specifies the number of bits to use for the generated keys.",
+			},
+			"key_usage": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Policies to be set on tokens issued using this role.",
+			},
+			"use_csr_common_name": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When used with the CSR signing endpoint, the common name in the CSR will be used instead of taken from the JSON data.",
+			},
+			"use_csr_sans": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "When used with the CSR signing endpoint, the subject alternate names in the CSR will be used instead of taken from the JSON data.",
+			},
+			"ou": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Specifies the OU (OrganizationalUnit) values in the subject field of issued certificates.",
+			},
+			"organization": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Specifies the O (Organization) values in the subject field of issued certificates.",
+			},
+			"generate_lease": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates issued/signed against this role will have Vault leases attached to them.",
+			},
+			"no_store": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "If set, certificates issued/signed against this role will not be stored in the storage backend.",
+			},
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the PKI backend to configure.",
+				ForceNew:    true,
+				Default:     "pki",
```

Comment:
I'm _fairly_ certain you saw someone (almost certainly me!) using this in other resources in the provider. I think we're trying to move away from setting a default backend, in favor of making it explicit, to avoid accidents. So I think this would be better as `Required: true` with no Default. Sorry for setting a bad example there.

Reply:
NP. Changed.